src/arrayMethodJoin.js

```diff
@@ -4,8 +4,44 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+    let separ = separator;
+    const length = this.length - 1;
+
+    if (typeof separ === 'undefined') {
```

Comment:
Your separator can't be undefined, you set default value for it (coma) at function declaration
This if / else statement redundant, delete it